### PR TITLE
[Design] Move a single class over to Workspaces.

### PR DESCRIPTION
**Pull request by NTaylorMullen**
- Had to modify the API a bit to massage the `TagHelperFactsService` into a state that didn't rely on VS.

@rynowak what do you think about this approach for supporting Roslyn's based services + VisualStudio's? I moved one of the more troublesome classes as an experiment.

### CODE REVIEW

File: src/Microsoft.CodeAnalysis.Razor.Workspaces/DefaultRazorSyntaxFactsService.cs [hunk 1]
```diff
@@ -139,21 +135,21 @@ public override IReadOnlyList<TagHelperSpan> GetTagHelperSpans(RazorSyntaxTree s
             return results;
         }
 
-        public override int? GetDesiredIndentation(RazorSyntaxTree syntaxTree, ITextSnapshot syntaxTreeSnapshot, ITextSnapshotLine line, int indentSize, int tabSize)
+        public override int? GetDesiredIndentation(RazorSyntaxTree syntaxTree, int previousLineEndIndex, Func<int, string> getLineContent, int indentSize, int tabSize)
```
Comment (rynowak) [inline src/Microsoft.CodeAnalysis.Razor.Workspaces/DefaultRazorSyntaxFactsService.cs]: This is probably an example of a case where we would just split this particular API out into its own service.

File: src/Microsoft.VisualStudio.LanguageServices.Razor/RazorSyntaxFactsServiceExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+using System;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.CodeAnalysis.Razor;
+using Microsoft.VisualStudio.Text;
+
+namespace Microsoft.VisualStudio.LanguageServices.Razor
+{
+    public static class RazorSyntaxFactsServiceExtensions
+    {
+        public static int? GetDesiredIndentation(
```
Comment (rynowak) [inline src/Microsoft.VisualStudio.LanguageServices.Razor/RazorSyntaxFactsServiceExtensions.cs]: ugh. no

File: src/Microsoft.CodeAnalysis.Razor.Workspaces/TagHelperSpan.cs [hunk 3]
```diff
@@ -6,7 +6,7 @@
 using Microsoft.AspNetCore.Razor.Language;
 using Microsoft.AspNetCore.Razor.Language.Legacy;
 
-namespace Microsoft.VisualStudio.LanguageServices.Razor
+namespace Microsoft.CodeAnalysis.Razor
```
Comment (rynowak) [inline src/Microsoft.CodeAnalysis.Razor.Workspaces/TagHelperSpan.cs]: These are breaking changes.

**Comment by rynowak** [comment]
Ah, I think I see what you're doing now. This is a small set of breaking changes.

Some of the details of this are mixed up but the strategy in general is right.

**Comment by NTaylorMullen** [comment]
> Ah, I think I see what you're doing now. This is a small set of breaking changes.
> 
> Some of the details of this are mixed up but the strategy in general is right.

Ya, it's a breaking change as far as namespace go but it doesn't require Alex to change how the underlying code works. Just makes him use a different namespace.

**Comment by NTaylorMullen** [comment]
Closing till we revisit this area.

**Pull request closed without merging**